### PR TITLE
Service metrics

### PR DESCRIPTION
**Pull request by songxiaosheng**
## What is the purpose of the change

support service interface level metrics,this metrics will no method 

image

we can use  rpc-level="SERVICE"  config to open it
```xml
<dubbo:metrics protocol="prometheus" enable-jvm="false" rpc-level="SERVICE" >
        <dubbo:histogram enabled="true"/>
    </dubbo:metrics>
```

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/event/RequestEvent.java [hunk 1]
```diff
@@ -58,10 +59,18 @@ public static RequestEvent toRequestEvent(ApplicationModel applicationModel, Str
...
                                               Invocation invocation, String side) {
         MethodMetric methodMetric = new MethodMetric(applicationModel, invocation);
+        RequestEvent requestEvent = getRequestEvent(applicationModel, appName, metricsDispatcher, collector, invocation,
+            side, methodMetric, MetricsSupport.getInterfaceName(invocation));
+        return requestEvent;
+    }
+
+
+
+    private static RequestEvent getRequestEvent(ApplicationModel applicationModel, String appName, MetricsDispatcher metricsDispatcher, DefaultMetricsCollector collector, Invocation invocation, String side, ServiceKeyMetric methodMetric, String serviceKey) {
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/event/RequestEvent.java:69]: This method of extraction seems unnecessary, there are too many parameters, and it is only used once.

File: dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/MethodMetric.java [hunk 2]
```diff
@@ -38,10 +41,23 @@ public class MethodMetric extends ServiceKeyMetric {
...
         super(applicationModel, MetricsSupport.getInterfaceName(invocation));
-        this.methodName = RpcUtils.getMethodName(invocation);
         this.side = MetricsSupport.getSide(invocation);
         this.group = MetricsSupport.getGroup(invocation);
         this.version = MetricsSupport.getVersion(invocation);
+        if (isServiceLevel(applicationModel)) {
+            this.methodName = "";
+            return;
+        }
+        this.methodName = RpcUtils.getMethodName(invocation);
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/MethodMetric.java:51]: Why does methodName need to be assigned an empty string， Will this be more concise?
if (!isServiceLevel(applicationModel)) {
        this.methodName = RpcUtils.getMethodName(invocation);      
 }

**Review by wxbty — APPROVED** [review]

File: dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/MethodMetric.java [hunk 3]
```diff
@@ -38,10 +41,17 @@ public class MethodMetric extends ServiceKeyMetric {
 
     public MethodMetric(ApplicationModel applicationModel, Invocation invocation) {
         super(applicationModel, MetricsSupport.getInterfaceName(invocation));
-        this.methodName = RpcUtils.getMethodName(invocation);
         this.side = MetricsSupport.getSide(invocation);
         this.group = MetricsSupport.getGroup(invocation);
         this.version = MetricsSupport.getVersion(invocation);
+        this.methodName = isServiceLevel(applicationModel) ? null : RpcUtils.getMethodName(invocation);
```
Comment (AlbumenJ) [inline dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/MethodMetric.java:47]: Should pre-check for `isServiceLevel`. Prevent cal it for each invocation.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/MethodMetric.java (hunk 3 above)
Comment (songxiaosheng) [reply dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/MethodMetric.java:47]: Thank you, this is a good suggestion

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/AggregateMetricsCollector.java [hunk 4]
```diff
@@ -97,6 +98,7 @@ public AggregateMetricsCollector(ApplicationModel applicationModel) {
                 this.enableRequest = Optional.ofNullable(aggregation.getEnableRequest()).orElse(true);
             }
         }
+        this.serviceLevel = MethodMetric.isServiceLevel(applicationModel);
```
Comment (AlbumenJ) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/AggregateMetricsCollector.java:101]: Move into `if (isCollectEnabled()` and `if (optional.isPresent())`

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/HistogramMetricsCollector.java [hunk 5]
```diff
@@ -69,6 +71,7 @@ public HistogramMetricsCollector(ApplicationModel applicationModel) {
 
             metricRegister = new HistogramMetricRegister(MetricsGlobalRegistry.getCompositeRegistry(applicationModel), histogram);
         }
+        this.serviceLevel = MethodMetric.isServiceLevel(applicationModel);
```
Comment (AlbumenJ) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/HistogramMetricsCollector.java:74]: Move into `if (config == null || config.getHistogram() == null || config.getHistogram().getEnabled() == null || Boolean.TRUE.equals(config.getHistogram().getEnabled()))`

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**